tests/e2e_tests/cypress/e2e/basic-test/commons.js

```diff
@@ -0,0 +1,88 @@
+const expectedPlace1 = {
+  title: "Grunwaldzki",
+  subtitle: "big bridge",
+  categories: [
+    ["type_of_place", "big bridge"],
+    ["accessible_by", "pedestrians, cars"]
+  ],
+  CTA: {
+    "type": "CTA",
+    "value": "https://www.example.com",
+    "displayValue": "Visit example.org!"
+  }
+};
+const expectedPlace2 = {
+  title: "Zwierzyniecka",
+  subtitle: "small bridge",
+  categories: [
+    ["type_of_place", "small bridge"],
+    ["accessible_by", "bikes, pedestrians"]
+  ]
+};
+
+export const expectedPlaces = [expectedPlace1, expectedPlace2];
+
+const zoomInTimes = 1;
+
+export function zoomInMap() {
+  for (let i = 0; i < zoomInTimes; i++) {
+    cy.get('.leaflet-marker-icon').first().click();
+    cy.wait(500);
+  }
+}
+
+// TODO - Find a way to search for a specific point, not iterate over all of them
+export function verifyArbitraryPopupContent(expectedPlaces) {
+  cy.get('.point-title').should('exist').invoke('text')
+    .then((title) => {
+      const expectedPlace1 = expectedPlaces[0];
+      const expectedPlace2 = expectedPlaces[1];
+      if (title === expectedPlace1.title) {
+        verifyPopupContent(expectedPlace1);
+      } else if (title === expectedPlace2.title) {
+        verifyPopupContent(expectedPlace2);
+      }
```

Comment:
I see what you did there. Well... that's naughty. We do not want tests to have logic. It should be deterministic.

Reply:
Not part of this PR. Resolving.